api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -616,8 +616,8 @@ public T map(Types.MapType map, T keyResult, T valueResult) {
       return null;
     }
 
-    public T variant() {
-      return null;
+    public T variant(Types.VariantType variant) {
+      throw new UnsupportedOperationException("Unsupported type: variant");
```

Comment:
This still need to be implemented by:
* `CreateMapping`
* `DescribeSchemaVisitor`
* `IdToOrcName` and other conversions

I think it's fine to implement the conversions later as part of Avro, Parquet, and ORC work, but the common ones should probably be implemented.

Reply:
Added `CreateMapping` and `DescribeSchemaVisitor` visitor implementation.  I will try to address the remaining from different modules separately.